### DISCUSSION TITLE
Allow "save and stay" to be the default save button action

### DISCUSSION THREAD

**Discussion by jordie23** [directus/directus#4081] [Feature Requests]
I'm currently evaluating Directus CMS for our site. 

While testing, I've noticed the save button on content is "save and return to list". I know this will annoy our writers, is there a way to default it to "save and stay"? I know there's command+s, but not all our writers are the most tech-aware and would opt for the button still. 

If it's not a feature already (couldn't see it anywhere) then I'd like to request it. Perhaps it could be a user preference option?

**Comment by benhaynes** [comment]
Would this be a Project, Role, User, or Collection setting?

**Reply by rijkvanzanten** [reply]
I _think_ this would be.. project? and/or role. It feels a little messy to do it per collection (inconsistent).

Per user could be nice too 🤔

**Reply by philleepflorence** [reply]
If configurable: **project**... **Role** would be nice, but not sure how different it would be from project, not encountered much of this.

If not configurable: Having the default match `CMD+S` or `CTRL+S` could help many editors used to that UX. This is a constant but minor issue with many new clients.

**Reply by rijkvanzanten** [reply]
> but not sure how different it would be from project

Project means globally for everybody, role is per role

**Reply by philleepflorence** [reply]
> Project means globally for everybody, role is per role

Oh I know...I meant that I have not seen where clients in different roles in the same project had different issues with saving...it was usually with projects.
The more granular the options, the easier of course!

**Reply by rijkvanzanten** [reply]
> The more granular the options, the easier of course!

Don't know if I agree with that! Having to configure it for every collection manually as a user becomes very cumbersome as well.

**Reply by benhaynes** [reply]
I vote for **global user preference only**. It defaults to what we have now, but users can override this with a default action (global, not per collection/role). We _could_ take that a step further and have a project default... but that's more of an add-on.

**Reply by jordie23** [reply]
> I vote for **global user preference only**. It defaults to what we have now, but users can override this with a default action (global, not per collection/role). We _could_ take that a step further and have a project default... but that's more of an add-on.

Basically how I envisaged it. Though I did think it could be something people could toggle in their own profile, a global default would at a minimum be very helpful.

**Reply by philleepflorence** [reply]
> I vote for global user preference only. It defaults to what we have now, but users can override this with a default action (global, not per collection/role). We could take that a step further and have a project default... but that's more of an add-on.

This works!

**Reply by Microphunk** [reply]
Also would be nice if you could Hide these "Special Actions"  , the ... button. 
And fields "Clear Field" and "Raw Value" seems quiet odd feature to me as well. Users just don't interact with apps like this. If they want to clear the field, they will focus on input and backspace it or select txt -> delete. (Visibility options) for these would be nice.

**Reply by benhaynes** [reply]
> Users just don't interact with apps like this.

Can you provide a but more insight regarding this? These features were added by request. Some interfaces are not as easy to clear (eg: a map, relational data, etc)... and with our support for _custom_ interfaces, we need to have fallback in case developers forget to include these key features.

Also, any reason the "..." save menu should have an option to be hidden? Adding random settings like this can make the platform far more complex to configure, with little benefit.

**Comment by insidesmart** [comment]
Is there any clean way to override the default save functionality via extensions until the feature is available in the product ?

**Reply by rijkvanzanten** [reply]
No, extensions currently don't have the ability to override the save button behavior in-app 👍🏻

**Reply by JonL1** [reply]
Another use case for refactoring Studio client/server logic into (scoped?) system-flows. Humongous refactoring project I know…but one can dream :) 

I can also imagine the bug triaging nightmare 😂

**Comment by rijkvanzanten** [comment]
Heya! Thanks for opening this feature request!

This feature request has received over **15 votes** from the community. This means we'll move this feature request to the Under Review state! 

The Core team will schedule a meeting to review this request as soon as possible. The discussion will then be approved or denied. You may or may not be invited to join this meeting with the core team.

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._

**Comment by rasco** [comment]
Has anyone started working on this issue?
I think just a global pref would suffice for us. 
We prefer the "Save and stay" option as default.

**Comment by rasco** [comment]
Now, with the [v10.2.0](https://github.com/directus/directus/releases/tag/v10.2.0) release having Preview Mode, the default "Save and leave" functionality makes less sense. 

Especially considering that the mod+S keyboard shortcut (at least on Mac/Chrome) often doesn't work correctly (it is sometimes captured by the browser which then tries to save the HTML page on disk).

As long as preview mode is enabled, I'd expect the default save functionality to save and refresh the preview pane, so I can view my changes.

**Reply by JonL1** [reply]
I wouldn’t even consider this a feature request. Just an understandable omission of the preview functionality.

**Reply by a-fazal** [reply]
I agree! Would be really helpful to Save and Stay in this case instead.

**Reply by rijkvanzanten** [reply]
> I agree! Would be really helpful to Save and Stay in this case instead.

Is it something you'd want to configure for the user, the role, per collection, or system wide? Alternatively, should it just default to whatever you used last?

**Reply by rasco** [reply]
I guess it should automatically be `Save and stay` in this case.
But everywhere else it could be a user setting with a configurable default.

**Reply by rasco** [reply]
Actually, after thinking about it, as another option, the `Save and stay` and `Save and close` buttons could be side-by-side, so you can reach whichever button you need right there.

**Reply by paescuj** [reply]
Applying different functionalities to the same button doesn't feel like good UX.
So I tend to both button being visible - ideally on all pages for consistency.

**Reply by rijkvanzanten** [reply]
> Applying different functionalities to the same button doesn't feel like good UX.
> So I tend to both button being visible - ideally on all pages for consistency.

I don't know if I agree there. There's 3 different save options + a discard changes option within that subnav; having all 4 as buttons side by side might be very confusing too

**Reply by rasco** [reply]
A part of the solution could be to make the different save buttons have different designs: 
The default one could be red, as it is now, while the 'Save and stay' one could be white inside with a thick red border.

**Reply by rijkvanzanten** [reply]
> The default one could be red, as it is now

Do you have custom CSS configured? By default, it's purple (primary) 👀

**Reply by rasco** [reply]
Ooops haha, yes, the default primary color I mean. Not red

**Reply by a-fazal** [reply]
Due to the preview functionality personally it makes most sense for me to set the save option by collection. However by user is also good with me! Whatever makes it easiest to start with

**Comment by rijkvanzanten** [comment]
Here's some ideas!

- We should split up "Save and X" from "Save as Copy" and "Revert Changes"
- Then we can default to main button to the "Save and X" operation you used last.
- We can store that in localstorage maybe? 
- I think it'd annoying to have to go to a different page to configure this, as it feels like something you might want to change on a case by case basis. 
- I don't think it should change the keyboard shortcuts
- It should change the icon + tooltip for whatever the current "default" action is

**Reply by rijkvanzanten** [reply]
~~One~~ two more things:

- Per collection or global?
- Save it to the user to persist?
- Make it per session?
- Not make it persist at all?
  - By that I mean as soon as you navigate away from the item detail page it resets back to the default default
  - ^ that feels good

**Reply by paescuj** [reply]
LOVE IT! How about a button similar to
<img width="200" src="https://github.com/directus/directus/assets/5363448/ef2bbc4f-9582-4337-a7aa-70103f72c59a">
?

**Reply by craigharman** [reply]
My vote would be for per collection.
Per user persistence would be great.

**Reply by u12206050** [reply]
Hmm wouldn't this make keyboard binding a bit confusing? I would expect CMD+S to be the default, and CMD+SHIFT+S to be the alternative.

**Comment by nalognl** [comment]
Any progress here ? 

We use Directus as CRM system and redirect after every client card or deal change is very annoying for users. 

Users are mainly regural managers and it is hard to explain to them about 3 dots option button, the just forget and complain about redirect. 

Lets make it per collections setting. 

Ideal will be to save and refresh page option becasue we use preview.

Such as it is not possible to change button via extention this feature is very urgent for us.

**Comment by rolodato** [comment]
To me, the facts that `CMD/Ctrl + S` is the shortcut for "Save and Stay", and that there is no keyboard shortcut for "Save and go back", are a clear signal that the default action should be Save and Stay. Users that mainly use the mouse or touch input (i.e. usually non-technical folks) are left out of this quick convenience, unfortunately.

I would be very curious to see a use case where Save and Stay should not be the default, I really can't think of any other application that behaves this way.

Also, the wording of the Save icon tooltip is deceiving given this default behavior. "Save" in every single application means just Save, and not doing anything else.

**Comment by vzkn** [comment]
I agree with @rolodato.

Our team is new to Directus, and we're we're loving it. However, we find the current saving logic a bit unintuitive for both technical users and those without a tech background. Given that our team primarily consists of content writers, they found it particularly challenging to adapt to using `CMD/Ctrl + S` for save and stay.

This issue becomes even more problematic when content writers are working on nested content. Accidental press of the save button leads to a redirection to the collection page of the nested element, which can be very confusing for them.

For my specific scenario, I would appreciate having two levels of control: one that's `global` and another that's `role-based`.

Many thanks to the Directus team for your hard work on such an excellent product!

**Comment by matiyin** [comment]
It's been more than 3 years since this topic started, but still nothing decided?

+1 for per-collection and role-based, otherwise global or 'remember last action' would already be an enormous improvement for editors. If you press ctrl+S when focused inside an input the browser save popups up instead of triggering a save in directus.

**Comment by figurethree** [comment]
+1 for save and stay by default.  We desperately need this to optimize the workflow when working with nested relationship items.

**Comment by jeremyzilar** [comment]
I wanted to add in our editors experiences with "saving" in Directus ☁️  

When an editor is working on a DRAFT article, they have a habit of regularly saving it as they go _(as a good practice)_. As stated already, this continually bumps the editor out of the article and into the list view. When you have a few hundred drafts, finding the specific article you were in is proving to be really challenging for our staff. 

- We discovered that you can do `command + s` and it will save your article (nice!), however, it only works if none of the fields are in focus. You have to click out of the field you are in, then do `command + s` for this to work.
- We didn't realize that there was a "save and stay" option under the ... till reading this thread
- When editing a "flow" save and stay editing seems to be the default.
- There is already a back to the list button on the left of the page
- Personally, I feel like you could make this switch in the core, then (if you need) provide a setting in the collection settings to change the behavior of the save button to go back to the list. I can see it being helpful to just exit the view upon saving when you are editing a more data-specific content type and don't need to preview the output _(like with articles)_.

**Comment by sbuchweitz** [comment]
please add SAVE & STAY as default option

**Comment by unknownterritory** [comment]
If nothing else, the shortcut `Ctrl+S` needs to be changed. This is the shortcut to save the page in most browsers and my experience is that, either on Chrome or Firefox, on Linux, the shortcut, as it is, **always** triggers the browser's save page routine and not Directus'.

I also agree that, especially now that "preview" is available, the default is for editors to want to save and stay.

This seems like a low-hanging fruit that could be addressed quickly and with great user benefit. I am a new user of Directus and this is such a surprising bug (it is truly a UI bug) that... here I am :-)

**Reply by u12206050** [reply]
It doesn't **always** trigger the page save, only when you have a textarea in focused. Clicking on the page outside any inputs or textarea and click `Cmd+S` works at least on my mac it does :)

**Reply by unknownterritory** [reply]
I develop on Linux, Pop!_OS. It happens **always**. Anyway, your observation is beside the point: it is not a good user experience to have to use the mouse for a keyboard shortcut to work :)

**Comment by figurethree** [comment]
Lately you have to click somewhere inside the right column (activity log,
etc) in order to use ctrl S.  Quite annoying.  Happened after the last
update.

**Comment by jasonsuriano** [comment]
We've been waiting for this option/ feature to be improved since 2021. 

It's such an issue for our users that we created a "Save" button that performs the Save and Stay function. 

We included the new button as part of a secondary navigation "banner" in each of our core editing sections. 

Here's a video of my Product Director showing how it works: 

https://www.dropbox.com/scl/fi/v8irwjtmm7le195009fpf/D11-Save-and-Stay.mp4?rlkey=a70n0bj4969nvepqogtgdsl6n&dl=0

If this is of interest, let me know, and we'd be happy to share the code/ steps to recreate.

**Reply by unknownterritory** [reply]
Love it!

**Reply by sbuchweitz** [reply]
thanks! I was getting ready to go down that route as well. 
I would very much appreciate if you could share your code and instructions.

**Reply by pedrofnts** [reply]
hey jason, I'm also interested on this!

**Reply by franforlizzi** [reply]
amazing! i'm interested.

**Reply by tomas-bara-meditorial** [reply]
Hi, I'm interested as well. Looks good, its jush shame it is not native ):

**Reply by SergeyRe** [reply]
Great ! Please describe way to realize it ( extension, flows...?)

**Reply by jasonsuriano** [reply]
Sorry for the delay everyone.

We're a small team and have been heads down pushing updates to our product quickify.ai

Since we have interest, we'll post the information next week.

**Reply by jasonsuriano** [reply]
Here's where you can find the save and stay information:

**1. Directus Marketplace -** search for Save and Stay. Here's what it looks like.
https://www.dropbox.com/s/6wpccgkhjdss6og/Screenshot%202024-10-31%20at%202.55.18%E2%80%AFPM.png?dl=0

**2. Bitbucket -** here's the link:
https://bitbucket.org/rocketfuelgames/directus-extension-save-and-stay-button/src/main/ 

Hope this helps!

**Reply by roonready** [reply]
Hi ! Thanks a lot.
PS just  note that it  is "Presentation and aliases" type  of field at marketplace  description ,  as there are no pics there to   clarify this detail

**Reply by franforlizzi** [reply]
Just did a quick test and it works as intended. Thank you very much ♥ @jasonsuriano

**Reply by jasonsuriano** [reply]
> Hi ! Thanks a lot. PS just note that it is "Presentation and aliases" type of field at marketplace description , as there are no pics there to clarify this detail

Note: Version 1.0.1 has been published with a clarification in the Get Started section of the readme file.

**Reply by franforlizzi** [reply]
Just did another test and it appears that the button remains disabled when creating a new item, even after making changes. It works only after saving the first time.

**Reply by Sothos** [reply]
Version 1.0.2 fixes the initial state issue

**Reply by sbuchweitz** [reply]
Save & Stay button does not work when editing an INLINE Item (in overlay sidepanel). Please fix if possible. Thanks for the good work.

**Reply by Sothos** [reply]
Due to the way Directus is designed, as far as I know it is not possible to save the item content in a drawer (the right-side panel) independently of its parent item. The Directus interface's top right checkmark button of the drawer doesn't actually save anything, it just closes the drawer. The drawer contents are only saved when the parent item's Save function is used, after the drawer has been closed. Because of this, the Save and Stay button remains disabled in a drawer. I agree this can look like a bug, or be confusing. I'm open to suggestions.

**Reply by rijkvanzanten** [reply]
> Due to the way Directus is designed, as far as I know it is not possible to save the item content in a drawer (the right-side panel) independently of its parent item

This is on purpose 🙂 I wanna make sure you can cancel out of the changes you've made by leaving e parent item. Having all nested relational changes apply immediately without the parent being saved was a pretty rough UX.

**Reply by pavelbales** [reply]
Hi @jasonsuriano,

Thank you for the amazing work on your Directus plugin! The save-and-stay functionality solves a big pain point for me, and it has significantly improved my workflow. I really appreciate all the effort you've put into making it happen.

I did notice a couple of small issues happening on Directus 11.3.5, though:

1. Whenever there’s a validation error or an error thrown in a flow script, the button stays stuck in the progress state (spinning wheel) and doesn’t revert back to the check icon.
2. Occasionally, when I click the button, it enters the progress state (spinning wheel), but the save action doesn’t actually execute. I have to click the button again to trigger the save properly.

Would it be possible for you to look into these? I’d be more than happy to help test or provide additional details if needed.

Thanks again for your hard work—your plugin has truly made a difference for me!

**Reply by jasonsuriano** [reply]
Hi @BaelishCZ, thanks for letting me know. We haven't upgraded to the most recent version of Directus so this may be why we haven't seen this issue. We're planning on upgrading in the coming weeks, and I'll post once we've investigated, and have a solution.

**Reply by pavelbales** [reply]
Hi @jasonsuriano,

Thank you so much for your reply. I really appreciate you looking into this. No rush—take your time with the upgrade. I’d be more than happy to test anything or help troubleshoot if needed. Thanks again for all the great work you’re doing!

**Reply by pavelbales** [reply]
Hi @jasonsuriano,

I hope you’re doing well!

I just wanted to follow up and see if there’s been any progress. No rush at all — I know these things take time.

If there’s anything I can do to help test or troubleshoot, please let me know.

Thank you again for all your work on this plugin!

**Comment by roonready** [comment]
Next ideas : "Save and stay and preview "  or "Save and stay and launch dynamic URL"

**Comment by MarkBurvs** [comment]
Just having a little cry that this is still open. Don't mind me. 😭

**Reply by pavelbales** [reply]
Same here @benhaynes 

Directus is really amazing, but thing like this drastically diminish its user experience ☹️

Thanks for all the great work you’re doing anyway!

**Comment by BorisKamp** [comment]
I would love an option where you can set the save behavior PER data model. For example

- one data modal I would like to `Save and stay`
- Other data model I would like to `Save and redirect to dynamic URL (frontend)` (for example: `https://frontend.com/blog-post/:id`)
Especially the latter one would be a really nice addition IMO!

**Comment by rafarc21** [comment]
A workaround is to use this extension: https://www.npmjs.com/package/@quickifyai/directus-extension-save-and-stay-button

**Comment by angelod1as** [comment]
Man, it's 2026 and this is still not solved... why did I choose directus :(

**Reply by rijkvanzanten** [reply]
haha it’s an opinion thing more so than a technical limitation :) Despite what you see in this thread, when exploring this with our product team we found it’s very close to 50/50 on whether or not this should be the default so we haven’t rolled it out org-wide. That being said, we’re doing a bit editorial project at the moment that includes global draft states and other versioning improvements which comes with some changes to the save button behavior as well, so rest assured: this isn’t completely ignored 😉

**Reply by MarkBurvs** [reply]
I agree that neither approach should be forced on anyone. Just give us the choice to set our own default at the project level 🙏. 

In our project, there is really no use-case for save-and-leave, so we have to train our users to use save and stay from the sub menu, or use the keyboard shortcut of cmd-s (which regularly stops working and triggers the browser's file-save instead which is super-frustrating).

**Reply by nalognl** [reply]
> haha it’s an opinion thing more so than a technical limitation :) Despite what you see in this thread, when exploring this with our product team we found it’s very close to 50/50 on whether or not this should be the default so we haven’t rolled it out org-wide. That being said, we’re doing a bit editorial project at the moment that includes global draft states and other versioning improvements which comes with some changes to the save button behavior as well, so rest assured: this isn’t completely ignored 😉

> In this thread, there was a good explanation of why by Rolodato:
> [https://github.com/directus/directus/discussions/4081#discussioncomment-7781784](https://github.com/directus/directus/discussions/4081#discussioncomment-7781784)

It would be interesting to hear why someone might disagree. While the “save and close” option may be required for some specific use cases - there’s no doubt that someone will always need something special.

**Reply by mitraxis** [reply]
There is never a use case for Save and Leave. Who made this decision. Why are we forced to leave a page while working on it every time we want to save content. Insane.

**Comment by mitraxis** [comment]
Why is this feature still not available?

**Comment by JamesW1** [comment]
Hi All 👋 

Since Directus [v12.2.0](https://github.com/directus/directus/releases) this can now be configured globally in Directus settings:
<img width="2142" height="1020" alt="image" src="https://github.com/user-attachments/assets/01c0ae2e-3e80-41fe-be30-38cdc87da188" />

Reference: https://github.com/directus/directus/pull/27993

**Reply by MarkBurvs** [reply]
Great news. Thanks @JamesW1
